### PR TITLE
Include hash of read-only manifest handle values in handle id.

### PR DESCRIPTION
**Pull request by shaper**
Early-ish checkpoint to solicit feedback. In particular:
- seeking to make sure we shouldn't be doing something to else to check read-only status such as scrutinizing any associated connection/type, I'm just presuming the spot where I'm computing and inserting the hash in the `unitType.isEntity` block within `_processStore` is one where that entity is implicitly always read-only, but perhaps I am naive/mistaken.
- the `handle._store.id` we discussed in chat is computed [here](https://github.com/PolymerLabs/arcs/compare/master...shaper:hash_read_only?expand=1#diff-7cf67f07396d399da8d0403d5df1dc2bR922) and is not currently updated to include the hash value, should it be? It seems a separate id to me, and then should the hash included be the hash of just that entity's value or the full list of entities (the latter is what's inserted into the handle id)?
- removed `immediate` from the id of particle specs since we're including hash instead which seems sufficient, but if we still want it, let's discuss, because then shouldn't we have it in the read-only entity case as well?
- this PR doesn't address the work item around turning off duplicate-id errors for read-only data, but this work should be sufficient for the Social muxing embedded recipe case where the game id value does always vary, so we can file an issue for the duplicate-id work and pursue separately

Part of https://github.com/PolymerLabs/arcs/issues/842

### CODE REVIEW

**Comment by shans** [comment]
* It looks to me like you're not computing an id for the entire store if the stored entities already have one serialized. This actually seems like a pretty good test of read-only-ness (if the entities have ids then they were written by something).

* the link you're providing for computation of handle._store.id is actually for entity ids. Definitely fine to ignore this one, but is it the one you mean?

* removing `immediate` is fine, I think

LMK about whether that is the code chunk you were thinking of for handle._store.id; otherwise I think this is looking good.

**Comment by shaper** [comment]
> LMK about whether that is the code chunk you were thinking of for handle._store.id; otherwise I think this is looking good.

TL;DR = I had `_store` and `_stored` mixed up, I think the PR is ok, our chat discussion are handle/store id had me double-checking, detail below.

=> Well, yes, with the current PR, running one of the tests added here and logging the entities when their id is computed [here](https://github.com/PolymerLabs/arcs/compare/master...shaper:hash_read_only?expand=1#diff-7cf67f07396d399da8d0403d5df1dc2bR922) in the case where `entity.$id` is not set:

```
%  ./tools/sigh test --grep='different ids'
[ { id: 'manifest:the.manifest::0', rawData: { nobId: '12345' } } ]
[ { id: 'manifest:the.manifest::0', rawData: { nobId: '67890' } } ]
```

Aiming to clarify what I was saying, originally I'd semi-blindly updated that id thinking it was one of the spots where handle id was computed, and then in looking into why the handle id wasn't in fact updated I found, as you note, that it's actually the entity id, and it turns out that entity ends up in `handle._stored` of the `InMemoryVariable` (actually `_stored` not `_store`, I mis-remembered), and so I removed the update to that id.

I think this PR is fine as-is based on your comment, I noted it since in chat it sounded like you were saying the store id should match the handle id, and in this case it doesn't unless we add something like, after line 922 in `manifest.js` in this PR:

```
        if (!entity.$id) {
          let entityHash = digest(JSON.stringify(entity));
          id = `${id}:${entityHash}`;
        }
```

which will then produce, for those entities:

```
[ { id: 'manifest:the.manifest::0:58f0de7504ffa8d8d1aa160e6bd2b168bfcd6b24',
    rawData: { nobId: '12345' } } ]
[ { id: 'manifest:the.manifest::0:0e71e39569f80e30bc88abf242b0a8c677d31327',
    rawData: { nobId: '67890' } } ]
```

**Comment by shaper** [comment]
Will merge, happy to revise further if new information comes to light...
